### PR TITLE
refactor: remove `Validation` from `Redundancy`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/8815

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Conflicted

**Comment by jaschdoc** [comment]
Merged

**Comment by magnus-madsen** [comment]
Conflicted again :D

**Comment by jaschdoc** [comment]
Merged

**Comment by jaschdoc** [comment]
Merged

**Comment by jaschdoc** [comment]
We will probably have one more merge conflict

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/api/Flix.scala [hunk 1]
```diff
@@ -527,7 +528,7 @@ class Flix {
       (_, patMatchErrors) = PatMatch.run(afterStratifier)
       redundancyErrors = Redundancy.run(afterStratifier)
       (_, safetyErrors) = Safety.run(afterStratifier)
-      errors = instanceErrors ++ stratificationErrors ++ patMatchErrors ++ redundancyErrors ++ safetyErrors
+      errors: List[CompilationMessage & Recoverable] = regionErrors ::: entryPointErrors ::: instanceErrors ::: stratificationErrors ::: patMatchErrors ::: redundancyErrors ::: safetyErrors
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Flix.scala]: Perhaps this line is getting too long :)

**Review by magnus-madsen — APPROVED** [review]

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/api/Flix.scala (hunk 1 above)
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/api/Flix.scala]: Yeah. I will fix it in another pr

**Pull request merged**